pkg/daemon/containernetwork/utils.go

```diff
@@ -235,3 +235,55 @@ func ListLocalAddressExceptLink(exceptLinkName string) ([]netlink.Addr, error) {
 func CheckIPIsGlobalUnicast(ip net.IP) bool {
 	return !ip.IsInterfaceLocalMulticast() && ip.IsGlobalUnicast()
 }
+
+func checkPodRuleExist(podCidr *net.IPNet, family int) (bool, error) {
+	ruleList, err := netlink.RuleList(family)
+	if err != nil {
+		return false, fmt.Errorf("list rule failed: %v", err)
+	}
+
+	for _, rule := range ruleList {
+		if rule.Src != nil && podCidr.String() == rule.Src.String() {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func checkPodNeighExist(podIP net.IP, forwardNodeIfIndex int, family int) (bool, error) {
+	neighList, err := netlink.NeighProxyList(forwardNodeIfIndex, family)
+	if err != nil {
+		return false, fmt.Errorf("list neighs for forward node if index %v failed: %v", forwardNodeIfIndex, err)
+	}
+
+	for _, neigh := range neighList {
+		if neigh.IP.Equal(podIP) {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func checkPodNetConfigReady(podIP net.IP, podCidr *net.IPNet, forwardNodeIfIndex int, family int) error {
```

Comment:
should we retry some times here to wait an async ready? I don't think it is good to let kubelet throw out the unnecessary errors.

Reply:
fixed